### PR TITLE
18 allow different input rasters for features

### PR DESCRIPTION
**Pull request by marcosbr**
Fix issue #18

### CODE REVIEW

File: predmap.py [hunk 1]
```diff
@@ -260,10 +263,18 @@ def clip(self, fname_out, fname_in):
...
-        """Transform raster to numpy array to be used for training and eval
         """
+            Transform raster to numpy array to be used for training and eval
+        """
+
         feats = []
+        aux = 0
+        self.list_of_features = []
         for raster in self.feature_rasters:
+            if aux == 0:
```
Comment (raplima) [inline predmap.py:274]:
```suggestion
```

File: predmap.py [hunk 2]
```diff
@@ -260,10 +263,18 @@ def clip(self, fname_out, fname_in):
...
+                self.dataframe = pd.DataFrame.from_records(itertools.product(range(raster.RasterYSize),
```
Comment (raplima) [inline predmap.py:275]:
```suggestion
```

File: predmap.py [hunk 3]
```diff
@@ -260,10 +263,18 @@ def clip(self, fname_out, fname_in):
...
+                                                            range(raster.RasterXSize)), columns=['Row', 'Column'])
```
Comment (raplima) [inline predmap.py:276]:
```suggestion
```

File: predmap.py [hunk 4]
```diff
@@ -274,9 +285,26 @@ def set_rasters_to_column(self):
                 band_np = band_np.astype(float)
                 # mark NaNs
                 band_np[band.GetMaskBand().ReadAsArray() == 0] = np.nan
-
                 feats.append(np.reshape(band_np, (-1, )))
 
+                # Separate rasters with single and multilayer
+                if raster.RasterCount == 1:
+                    path = self.fnames_features[aux]
+                    colname = path.split('\\')[-1].split('.')[0]
```
Comment (raplima) [inline predmap.py:293]: Better to use the `Path` commands. Something like:
````python
Path(path).resolve().stem
````

Also, the  outer loop can be zipped so no `aux` variables are necessary, maybe something like

````python
        for raster, raster_name in zip(self.feature_rasters, self.fnames_features):
             ...
````

**Review by marcosbr — COMMENTED** [review]
Agreed

File: predmap.py [hunk 5]
```diff
@@ -74,6 +72,11 @@ def __init__(self,
         self.int_to_lab = None
         self.target_raster_fname = None
         self.fname_lab_conv = None
+        self.list_of_features = None
+        self.dataframe = None
+        self.run_pca = True
+        self.list2pca = []
+       # self.nan_mask = None
```
[the remark this answers is not in the document]
Comment (marcosbr) [reply predmap.py:79]: I just forgot to declare

**Review by marcosbr — COMMENTED** [review]
agrred

**Review by marcosbr — COMMENTED** [review]
Agreed

**Review by marcosbr — COMMENTED** [review]
Agreed

**Review by marcosbr — COMMENTED** [review]
agreed

File: predmap.py (hunk 4 above)
Comment (marcosbr) [reply predmap.py:292]: Added the new outer loop and the suggestion of Path  to get the colnames. The code is more clean now1

**Pull request merged**